From 8e91e23fcf66735346f65fbf3dc3e82b85748cda Mon Sep 17 00:00:00 2001
From: Luo Jie <quic_luoj@quicinc.com>
Date: Tue, 12 Mar 2024 17:49:05 +0800
Subject: [PATCH 626/636] dt-bindings: net: Document Qualcomm QCA8386 PCS
 device

There are two PCS devices in QCA8386, both PCSes support SGMII
and 2500BASE-X.

Signed-off-by: Luo Jie <quic_luoj@quicinc.com>
---
 .../bindings/net/pcs/qcom,qca8k-pcs.yaml      | 86 +++++++++++++++++++
 1 file changed, 86 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/net/pcs/qcom,qca8k-pcs.yaml

diff --git a/Documentation/devicetree/bindings/net/pcs/qcom,qca8k-pcs.yaml b/Documentation/devicetree/bindings/net/pcs/qcom,qca8k-pcs.yaml
new file mode 100644
index 000000000000..f89bdc03fa16
--- /dev/null
+++ b/Documentation/devicetree/bindings/net/pcs/qcom,qca8k-pcs.yaml
@@ -0,0 +1,86 @@
+# SPDX-License-Identifier: GPL-2.0-only OR BSD-2-Clause
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/net/pcs/qcom,qca8k-pcs.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Qualcomm QCA8386 PCS
+
+maintainers:
+  - Luo Jie <quic_luoj@quicinc.com>
+
+description:
+  QCA8386 have two PCS devices named as PCS0 and PCS1, which support
+  the interface mode SGMII and 2500BASE-X to connect with MAC5 and MAC0.
+  They are accessiable over the Ethernet MDIO bus.
+
+properties:
+  compatible:
+    const: qcom,qca8k-pcs
+
+  reg:
+    maxItems: 1
+
+  clocks:
+    items:
+      - description: PCS system clock
+      - description: PCS RX clock
+      - description: PCS TX clock
+
+  clock-names:
+    items:
+      - const: sys
+      - const: rx
+      - const: tx
+
+  resets:
+    items:
+      - description: PCS RX reset
+      - description: PCS TX reset
+
+  reset-names:
+    items:
+      - const: rx
+      - const: tx
+
+  '#clock-cells':
+    const: 1
+
+  clock-output-names:
+    maxItems: 2
+
+required:
+  - compatible
+  - reg
+  - clocks
+  - clock-names
+  - resets
+  - reset-names
+  - '#clock-cells'
+  - clock-output-names
+
+additionalProperties: false
+
+examples:
+  - |
+    #include <dt-bindings/clock/qcom,qca8k-nsscc.h>
+    #include <dt-bindings/reset/qcom,qca8k-nsscc.h>
+
+    mdio {
+        #address-cells = <1>;
+        #size-cells = <0>;
+        pcs1@6 {
+            compatible = "qcom,qca8k_pcs";
+            reg = <6>;
+            clocks = <&qca8k_nsscc NSS_CC_SRDS1_SYS_CLK>,
+                     <&qca8k_nsscc NSS_CC_MAC0_RX_SRDS1_CLK>,
+                     <&qca8k_nsscc NSS_CC_MAC0_TX_SRDS1_CLK>;
+            clock-names = "sys", "rx", "tx";
+            resets = <&qca8k_nsscc NSS_CC_MAC0_RX_SRDS1_ARES>,
+                     <&qca8k_nsscc NSS_CC_MAC0_TX_SRDS1_ARES>;
+            reset-names = "rx", "tx";
+            #clock-cells = <0x1>;
+            clock-output-names = "qca8k_uniphy_rx",
+                                 "qca8k_uniphy_tx";
+        };
+    };
-- 
2.34.1

